src/ql/com/dec/unitary.cc

```diff
@@ -665,6 +666,14 @@ static Int recursiveRelationsForUnitaryDecomposition(
     }
 }
 
+/**
+ * Does state preparation, results in a circuit for which A|0> = |psi> for given state psi, stored as the array in Unitary
+ */
+ir::compat::GateRefs Unitary::prepare_state(const utils::Vec<utils::UInt> &qubits){
+    QL_EOUT("State preparation not implemented!");
```

Comment:
Since you do a PR with multiple commits, you might as well squash all the commits together and clean up so that the history of the main branch is clean after the merge.